pyrasterframes/python/geomesa_pyspark/types.py

```diff
@@ -29,40 +31,64 @@ def scalaUDT(cls):
         return 'org.apache.spark.sql.jts.' + cls.__name__
 
     def serialize(self, obj):
-        if (obj is None): return None
-        return Row(obj.toBytes)
+        return [_serialize_to_wkb(obj)]
 
     def deserialize(self, datum):
-        return RFContext._jvm_mirror().generate_geometry(datum[0])
+        return _deserialize_from_wkb(datum[0])
 
 
-class PointUDT(GeometryUDT):
+class PointUDT(ShapelyGeometryUDT):
     pass
 
 
-class LineStringUDT(GeometryUDT):
+class LineStringUDT(ShapelyGeometryUDT):
     pass
 
 
-class PolygonUDT(GeometryUDT):
+class PolygonUDT(ShapelyGeometryUDT):
     pass
 
 
-class MultiPointUDT(GeometryUDT):
+class MultiPointUDT(ShapelyGeometryUDT):
     pass
 
 
-class MultiLineStringUDT(GeometryUDT):
+class MultiLineStringUDT(ShapelyGeometryUDT):
     pass
 
 
-class MultiPolygonUDT(GeometryUDT):
+class MultiPolygonUDT(ShapelyGeometryUDT):
     pass
 
 
-class GeometryUDT(GeometryUDT):
+class GeometryUDT(ShapelyGeometryUDT):
     pass
 
 
-class GeometryCollectionUDT(GeometryUDT):
+class GeometryCollectionUDT(ShapelyGeometryUDT):
     pass
+
+
+def _serialize_to_wkb(data):
+    if isinstance(data, BaseGeometry):
+        return bytearray(data.wkb)  # bytearray(...) needed for Python 2 compat.
+    return None
+
+
+def _deserialize_from_wkb(data):
+    if data is None:
+        return None
+    return wkb.loads(bytes(data))  # bytes(...) needed for Python 2 compat.
+
+
+_deserialize_from_wkb.__safe_for_unpickling__ = True
+
+# inject some PySpark constructs into Shapely's geometry types
+Point.__UDT__ = PointUDT()
+MultiPoint.__UDT__ = MultiPointUDT()
+LineString.__UDT__ = LineStringUDT()
+MultiLineString.__UDT__ = MultiLineStringUDT()
+Polygon.__UDT__ = PolygonUDT()
+MultiPolygon.__UDT__ = MultiPolygonUDT()
+BaseGeometry.__UDT__ = GeometryUDT()
+GeometryCollection.__UDT__ = GeometryCollectionUDT()
```

Comment:
Did you determine if GeoMesa was publishing anything via Pip or similar? My preference would be to get this behavior by adding `geomesa-pyspark` to `setup.py` and calling it a day... not possible?

If not, and this class is effectively a copy of the GeoMesa one, we should change the copyright accordingly.

Reply:
They do not publish to pypi for pip install etc. It is possible to pip install local artifact e.g. from maven central.  Not sure if that fits the bill for deployment of raster frames though. Or

Indeed this is a copy of teh GeoMesa one, and the copyright reflects that, as it was before.

Reply:
@vpipkt Ooops.... sorry I just looked at the top of the comment block and saw "Astraea", but didn't actually read it 🤦‍♂️